tutorials/sphinx_tuto/tensordict_keys.py

```diff
@@ -0,0 +1,171 @@
+# -*- coding: utf-8 -*-
+"""
+Manipulating the keys of a TensorDict
+=====================================
+**Author**: `Tom Begley <https://github.com/tcbegley>`_
+
+In this tutorial you will learn how to work with and manipulate the keys in a
+``TensorDict``, including getting and setting keys, iterating over keys, manipulating
+nested values, and flattening the keys.
+"""
+
+##############################################################################
+# Setting and getting keys
+# ------------------------
+# We can set and get keys using the same syntax as a Python ``dict``
+
+# sphinx_gallery_start_ignore
+import warnings
+
+warnings.filterwarnings("ignore")
+# sphinx_gallery_end_ignore
+import torch
+from tensordict.tensordict import TensorDict
+
+tensordict = TensorDict({}, [])
+
+# set a key
+a = torch.rand(10)
+tensordict["a"] = a
+
+# retrieve the value stored under "a"
+assert tensordict["a"] is a
+
+###############################################################################
+# .. note::
+#
+#    Unlike a Python ``dict``, all keys in the ``TensorDict`` must be strings. However
+#    as we will see, it is also possible to use tuples of strings to manipulate nested
+#    values.
+#
+# We can also use the methods ``.get()`` and ``.set`` to accomplish the same thing.
+
+tensordict = TensorDict({}, [])
+
+# set a key
+a = torch.rand(10)
+tensordict.set("a", a)
+
+# retrieve the value stored under "a"
+assert tensordict.get("a") is a
+
+###############################################################################
+# Like ``dict``, we can provide a default value to ``get`` that should be returned in
+# case the requested key is not found.
+
+assert tensordict.get("banana", a) is a
```

Comment:
maybe let's talk about set_default here?
IIRC we used an underscore when dicts don't, we should correct that

Reply:
I can submit a PR for that.

Reply:
https://github.com/pytorch-labs/tensordict/pull/235